Use the OGRE_NEXT variable to rename the libraries.

This avoids a filesystem conflict between OGRE and OGRE-Next lirbaries

Upstream-Status: Pending

Signed-off-by: Rob Woolley <rob.woolley@windriver.com>

Index: git/CMake/Templates/OGREConfig.cmake.in
===================================================================
--- git.orig/CMake/Templates/OGREConfig.cmake.in
+++ git/CMake/Templates/OGREConfig.cmake.in
@@ -21,7 +21,7 @@ include(FindPackageMessage)
 set(OGRE_PREFIX_DIR "@CMAKE_INSTALL_PREFIX@")
 get_filename_component(OGRE_LIBRARY_DIRS "${OGRE_PREFIX_DIR}/lib" ABSOLUTE)
 get_filename_component(OGRE_INCLUDE_DIRS "${OGRE_PREFIX_DIR}/include/OGRE" ABSOLUTE)
-set(OGRE_LIBRARIES "OgreMain")
+set(OGRE_LIBRARIES "@OGRE_NEXT@Main")
 
 message(STATUS "Found OGRE")
 message(STATUS "  libraries : '${OGRE_LIBRARIES}' from ${OGRE_LIBRARY_DIRS}")
Index: git/Tests/CMakeLists.txt
===================================================================
--- git.orig/Tests/CMakeLists.txt
+++ git/Tests/CMakeLists.txt
@@ -113,7 +113,7 @@ if (OGRE_BUILD_TESTS)
       include_directories(${CMAKE_CURRENT_SOURCE_DIR}/Components/Paging/include)
       ogre_add_component_include_dir(Paging)
 
-      set(OGRE_LIBRARIES ${OGRE_LIBRARIES} OgrePaging)
+      set(OGRE_LIBRARIES ${OGRE_LIBRARIES} ${OGRE_NEXT}Paging)
       list(APPEND HEADER_FILES Components/Paging/include/PageCoreTests.h)
       list(APPEND SOURCE_FILES Components/Paging/src/PageCoreTests.cpp)
     endif ()
@@ -121,7 +121,7 @@ if (OGRE_BUILD_TESTS)
       include_directories(${CMAKE_CURRENT_SOURCE_DIR}/Components/MeshLodGenerator/include)
       ogre_add_component_include_dir(MeshLodGenerator)
 
-      set(OGRE_LIBRARIES ${OGRE_LIBRARIES} OgreMeshLodGenerator)
+      set(OGRE_LIBRARIES ${OGRE_LIBRARIES} ${OGRE_NEXT}MeshLodGenerator)
       list(APPEND HEADER_FILES Components/MeshLodGenerator/include/MeshLodTests.h)
       list(APPEND SOURCE_FILES Components/MeshLodGenerator/src/MeshLodTests.cpp)
     endif ()
@@ -129,7 +129,7 @@ if (OGRE_BUILD_TESTS)
       include_directories(${CMAKE_CURRENT_SOURCE_DIR}/Components/Terrain/include)
       ogre_add_component_include_dir(Terrain)
 
-      set(OGRE_LIBRARIES ${OGRE_LIBRARIES} OgreTerrain)
+      set(OGRE_LIBRARIES ${OGRE_LIBRARIES} ${OGRE_NEXT}Terrain)
       list(APPEND HEADER_FILES Components/Terrain/include/TerrainTests.h)
       list(APPEND SOURCE_FILES Components/Terrain/src/TerrainTests.cpp)
     endif ()
@@ -137,7 +137,7 @@ if (OGRE_BUILD_TESTS)
       include_directories(${CMAKE_CURRENT_SOURCE_DIR}/Components/Property/include
         ${OGRE_SOURCE_DIR}/Components/Property/include)
 
-      set(OGRE_LIBRARIES ${OGRE_LIBRARIES} OgreProperty)
+      set(OGRE_LIBRARIES ${OGRE_LIBRARIES} ${OGRE_NEXT}Property)
       list(APPEND HEADER_FILES Components/Property/include/PropertyTests.h)
       list(APPEND SOURCE_FILES Components/Property/src/PropertyTests.cpp)
     endif ()
@@ -265,7 +265,7 @@ if (OGRE_BUILD_TESTS)
       include_directories(${CMAKE_CURRENT_SOURCE_DIR}/Components/MeshLodGenerator/include)
       ogre_add_component_include_dir(MeshLodGenerator)
 
-      set(OGRE_LIBRARIES ${OGRE_LIBRARIES} OgreMeshLodGenerator)
+      set(OGRE_LIBRARIES ${OGRE_LIBRARIES} ${OGRE_NEXT}MeshLodGenerator)
       list(APPEND HEADER_FILES Components/MeshLodGenerator/include/MeshLodTests.h)
       list(APPEND SOURCE_FILES Components/MeshLodGenerator/src/MeshLodTests.cpp)
     endif ()
